Source/Dafny/Resolver.cs

```diff
@@ -16985,6 +16980,20 @@ class MethodCallInformation {
       bindings.AcceptArgumentExpressionsAsExactParameterList(actuals);
     }
 
+    private static string GetLocationInformation(Formal formal, int formalsCount, int index) {
```

Comment:
Did you consider having the signature be `GetLocationInformation(IReadonlyList<Formal> formals, int index)` ?

Reply:
Yes I did consider the two. However, since we already extracted the formal previously from the enumerable, I don't think we should extract it one more time. Some collections can be enumerated only once.

Reply:
If you make the argument `IReadonlyList<Formal> formals` then you know the collection supports constant time random access so there's no downside in accessing the element twice.

Reply:
Indeed, that would work. However, with the `IReadonlyList<Formal>` we would further need to check that the index is within the bounds of the list, which we don't with the current method, so that's what I prefer. Despite that, would you still prefer the `IReadOnlyList<Formal>` version ?

Reply:
> we would further need to check that the index is within the bounds of the list

I don't think we should do that check. The caller is responsible for the index being in bounds, and in our two cases that would be guaranteed.

Reply:
Done